dlt/common/data_writers/writers.py

```diff
@@ -398,23 +402,32 @@ def __init__(
         *,
         delimiter: str = ",",
         include_header: bool = True,
+        quoting: CsvQuoting = "quote_needed",
         bytes_encoding: str = "utf-8",
     ) -> None:
         super().__init__(f, caps)
         self.include_header = include_header
         self.delimiter = delimiter
+        self.quoting: CsvQuoting = quoting
         self.writer: csv.DictWriter[str] = None
         self.bytes_encoding = bytes_encoding
 
     def write_header(self, columns_schema: TTableSchemaColumns) -> None:
         self._columns_schema = columns_schema
+        if self.quoting == "quote_needed":
+            quoting: Literal[1, 2] = csv.QUOTE_NONNUMERIC
```

Comment:
`csv.QUOTE_NONNUMERIC` has the value of `2` and `1` corresponds to `csv.QUOTE_ALL` should we change this to int from `Literal`?

Reply:
so I did a read up I thought, type information would be erased and become any in the case of `else` branch but according to mypy it reads the first type definition for a variable and later uses it to validate the assigned type.